compiler/src/Language/Bond/Codegen/TypeMapping.hs

```diff
@@ -513,10 +513,10 @@ javaType BT_Blob = pure "byte[]"
 javaType (BT_IntTypeArg x) = pureText x
 javaType (BT_Maybe type_) = javaType (BT_Nullable type_)
 javaType (BT_Nullable element) = javaBox element
-javaType (BT_List element) = "java.util.LinkedList<" <>> javaBox element <<> ">"
-javaType (BT_Vector element) = "java.util.ArrayList<" <>> javaBox element <<> ">"
-javaType (BT_Set element) = "java.util.HashSet<" <>> javaBox element <<> ">"
-javaType (BT_Map key value) = "java.util.HashMap<" <>> javaBox key <<>> ", " <>> elementTypeName value <<> ">"
+javaType (BT_List element) = "java.util.List<" <>> javaBox element <<> ">"
+javaType (BT_Vector element) = "java.util.List<" <>> javaBox element <<> ">"
+javaType (BT_Set element) = "java.util.Set<" <>> javaBox element <<> ">"
+javaType (BT_Map key value) = "java.util.Map<" <>> javaBox key <<>> ", " <>> elementTypeName value <<> ">"
```

Comment:
Could you please clarify why you made this change? I thought it's driven by compile-type switch but defaults to concrete collection types.